galaxy_ng/app/migrations/0003_synclist_upstream_and_migrate.py

```diff
@@ -0,0 +1,60 @@
+# Generated by Django 2.2.13 on 2020-07-13 18:22
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+def set_default_upstream_repository(apps, schema_editor):
+    AnsibleRepository = apps.get_model('ansible', 'AnsibleRepository')
+    SyncList = apps.get_model('galaxy', 'SyncList')
+
+    db_alias = schema_editor.connection.alias
+
+    synclists = SyncList.objects.using(db_alias).all()
+
+    # Nothing to update
+    if not synclists:
+        return
+
+    try:
+        default_repo = AnsibleRepository.objects.using(db_alias).get(name="automation-hub")
+    except AnsibleRepository.DoesNotExist:
+        # If there are synclists, but no default repo, then let the exception raise
+        if synclists:
+            raise
```

Comment:
Probably makes sense to raise more descriptive error rather than `DoesNotExist`, because with this check the error starts having different meaning.